ReactWindows/ReactNative/Views/DatePicker/ReactCalendarDatePickerManager.cs

```diff
@@ -0,0 +1,232 @@
+using System;
+using Windows.UI.Xaml.Controls;
+using ReactNative.UIManager;
+using ReactNative.UIManager.Annotations;
+using ReactNative.UIManager.Events;
+using Newtonsoft.Json.Linq;
+
+namespace ReactNative.Views.DatePicker
+{
+    /// <summary>
+    /// A view manager responsible for rendering a CalendarDatePicker.
+    /// </summary>
+    public class ReactCalendarDatePickerManager : BaseViewManager<CalendarDatePicker, ReactCalendarDatePickerShadowNode>
+    {
+        /// <summary>
+        /// The name of the view manager.
+        /// </summary>
+        public override string Name
+        {
+            get
+            {
+                return "CalendarDatePickerWindows";
+            }
+        }
+
+        /// <summary>
+        /// Sets the value of the picker.
+        /// </summary>
+        /// <param name="view">The picker view element.</param>
+        /// <param name="date">The new value.</param>
+        [ReactProp("date")]
+        public void SetDate(CalendarDatePicker view, DateTime? date)
```

Comment:
I'm surprised that DateTime is an accepted ReactProp argument type. My concern, though, is the user experience when dealing with the `Date` object in JavaScript. Most bridge APIs, that I've seen, that leverage the `Date` object from JavaScript serialize the date as a `long`, representing milliseconds since the epoch.

Reply:
I am not entirely sure whether that is a good idea. If the serialization works well, adding an extra layer of serialization and deserialization (Date->long->DateTime and then back) is a bit redundant, in my opinion. What would be the benefit for a developer that uses this component or for the codebase itself?

Reply:
I guess I'm curious why Android does [this](https://github.com/facebook/react-native/blob/8dc3ba0444c94d9bbb66295b5af885bff9b9cd34/Libraries/Components/DatePickerAndroid/DatePickerAndroid.android.js) for JS `Date` objects.

Reply:
Note - same goes for [iOS](https://github.com/facebook/react-native/blob/8dc3ba0444c94d9bbb66295b5af885bff9b9cd34/Libraries/Components/DatePicker/DatePickerIOS.ios.js#L150-L156)

Reply:
AFAIK - the only primitives supported as prop types are boolean, int, float, double, and string, so if this is working, it's likely because under the hood JS is converting the Date object to a string and Newtonsoft.Json happens to parse it correctly. This would end up being an "undocumented feature" that we support dates in this way, and I'd really prefer to avoid that (e.g., in case in the future we change the native prop binding mechanism from Newtonsoft.Json to something else).

Reply:
It seems to work! I only need to modify the `TimePickerWindows` component, the other two are good to go.